kubespawner/objects.py

```diff
@@ -353,41 +362,74 @@ def make_pod(
-    # the volumes as compared to the running user of the container. Volumes
-    # belong to the pod and are only mounted by containers after all.
+    # | Dedicated KubeSpawner API  | Kubernetes API           | Security contexts |
+    # | -------------------------- | ------------------------ | ----------------- |
+    # | supplemental_gids          | supplementalGroups       | Pod only          |
+    # | fs_gid                     | fsGroup                  | Pod only          |
+    # | -                          | fsGroupChangePolicy      | Pod only          |
+    # | -                          | sysctls                  | Pod only          |
+    # | privileged                 | privileged               | Container only    |
+    # | allow_privilege_escalation | allowPrivilegeEscalation | Container only    |
+    # | -                          | capabilities             | Container only    |
+    # | -                          | procMount                | Container only    |
+    # | -                          | readOnlyRootFilesystem   | Container only    |
+    # | uid                        | runAsUser                | Pod and Container |
+    # | gid                        | runAsGroup               | Pod and Container |
+    # | -                          | runAsNonRoot             | Pod and Container |
+    # | -                          | seLinuxOptions           | Pod and Container |
+    # | -                          | seccompProfile           | Pod and Container |
+    # | -                          | windowsOptions           | Pod and Container |
     #
     # ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
     # ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#securitycontext-v1-core (container)
     # ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#podsecuritycontext-v1-core (pod)
-    pod_security_context = V1PodSecurityContext()
+    #
+    psc = {}
+    # populate with fs_gid / supplemental_gids
     if fs_gid is not None:
-        pod_security_context.fs_group = int(fs_gid)
-    if supplemental_gids is not None and supplemental_gids:
-        pod_security_context.supplemental_groups = [
-            int(gid) for gid in supplemental_gids
-        ]
-    # Only clutter pod spec with actual content
-    if not all([e is None for e in pod_security_context.to_dict().values()]):
-        pod.spec.security_context = pod_security_context
+        psc["fsGroup"] = int(fs_gid)
+    if supplemental_gids is not None:
+        psc["supplementalGroups"] = [int(gid) for gid in supplemental_gids]
+    if pod_security_context is not None:
+        if any("_" in key for key in pod_security_context.keys()):
+            raise ValueError(
+                "pod_security_context's keys should be directly k8s compliant camelCase, but snake_case formatting was detected."
+            )
```

Comment:
When a value is wrong, it's a good idea to show that value so it's easier to find and fix. For example:

```suggestion
        for key in pod_security_context.keys():
            if "_" in key:
                raise ValueError(
                    f"pod_security_context's keys should have k8s camelCase names, got '{key}'"
                )
```